src/query/frontend/semantic/symbol_generator.cpp

```diff
@@ -160,8 +161,18 @@ void SymbolGenerator::VisitReturnBody(ReturnBody &body, Where *where) {
 // CypherQuery
 
 bool SymbolGenerator::PreVisit(CypherQuery &cypher_query) {
-  if (cypher_query.pre_query_directives_.commit_frequency_) {
-    throw utils::NotYetImplemented("periodic commit");
+  bool const has_commit_frequency = !!cypher_query.pre_query_directives_.commit_frequency_;
+
+  if (global_scope_.has_periodic_commit && has_commit_frequency) {
+    throw SemanticException("You can specify periodic commit only once during a query!");
+  }
```

Comment:
commit frequency is an expression, I can't do it like this, since at this point I don't have it converted to int. 
And I don't want to involve expression evaluator here since that involves typed values and similar stuff which we haven't done yet.

Reply:
Sorry, I was dumb, didn't change the comment once I checked code. Maybe you can do here same as with others:
`bool const has_commit_frequency = cypher_query.pre_query_directives_.commit_frequency_ != nullptr;`